lib/OperatingSystems/Linux.php

```diff
@@ -139,12 +143,17 @@ public function getUptime(): int {
 	}
 
 	public function getNetworkInfo(): array {
-		$result = [];
-		$result['hostname'] = \gethostname();
-		$dns = shell_exec('cat /etc/resolv.conf |grep -i \'^nameserver\'|head -n1|cut -d \' \' -f2');
-		$result['dns'] = $dns;
-		$gw = shell_exec('ip route | awk \'/default/ { print $3 }\'');
-		$result['gateway'] = $gw;
+		$result = [
+			'hostname' => \gethostname(),
+			'dns' => '',
+			'gateway' => '',
+		];
+
+		if (function_exists('shell_exec')) {
```

Comment:
I guess the reason we can't use `executeCommand()` here is because of the inherent use of `escapeshellcmd()`. Maybe we add an optional parameter to `executeCommand()  that specifies whether to escape or not (that defaults true). We could then use consistently (elsewhere too I think) w/o boilerplate.